### PR TITLE
Add basic HiDPI support

### PR DESCRIPTION
**Pull request by tintou**

### CODE REVIEW

**Comment by tintou** [comment]
Note that this doesn't cover all the cases but is a preliminary work on HiDPI

File: src/asgen/utils.d [hunk 1]
```diff
@@ -47,33 +47,53 @@ struct ImageSize
...
-        return height;
+            return width * scale;
+        return height * scale;
     }
 
     int opCmp (const ImageSize s) const
     {
+        if (this.scale != s.scale) {
+            if (this.scale > s.scale)
+                return 1;
```
Comment (ximion) [inline src/asgen/utils.d]: This doesn't look right... That was an 32x32@2 size will be ordered before an 128x128 size. I don't think this has practical implications, but we should likely resolve that so nobody wonders about this in future.

**Comment by ximion** [comment]
Looks pretty good - the tarball generation is missing, but that's trivial to do.

Did you test this and see if icons are correctly generated?

**Comment by tintou** [comment]
Should be fixed, I'll try to generate the files on the elemntary PPA once I get some time

File: src/asgen/handlers/iconhandler.d [hunk 2]
```diff
@@ -129,6 +136,9 @@ public:
 
     private bool directoryMatchesSize (Algebraic!(int, string)[string] themedir, ImageSize size)
     {
+        int scale = themedir["scale"].get!(int);
```
Comment (ximion) [inline src/asgen/handlers/iconhandler.d]: You could make this just `themedir["scale"].get!int;`, D doesn't need the brackets here (that particular code is rather old from when I didn't know that much about D yet ^^)
But that's just a style-thing, no need to change it.

File: src/asgen/engine.d [hunk 3]
```diff
@@ -320,10 +320,13 @@ public:
 
         // prepare icon-tarball array
         immutable iconTarSizes = ["64", "128"];
+        immutable iconTarScales = ["1", "2"];
```
Comment (ximion) [inline src/asgen/engine.d]: I wonder whether we should iterate over `wantedIconSizes` here, instead of defining another array (minor thing though).

**Comment by ximion** [comment]
Still looking good, in fact I think this is merge-ready. I'd just like to know whether you tested this meanwhile (some additional data on this is helpful).

**Comment by ximion** [comment]
Hmm, it looks like I forgot to merge this... Sorry for the delay! I'll merge it and resolve a few minor quirks later.

**Pull request merged**